### PR TITLE
Ensure settings editor is attached before activation

### LINKED ISSUE #12485: JupyterLab Advanced (JSON) settings editor sometimes does not launch
## Description

The JupyterLab JSON advanced settings editor sometimes does not launch.

## Reproduce

1. Install JupyterLab 3.3.4 (macos 12, m1 processor)
2. Open command palette and execute the "Advanced JSON settings editor" command

Sometimes the JSON editor does not appear. For me, in launching jupyterlab 5 times, the command does not work maybe in one of the launches. If the command doesn't work the first time, it seems to not work until jupyterlab is relaunched. Refreshing the page doesn't seem to fix it.

There may be a race condition of some kind? This is puzzling.

This came up in the PyCon US 2022 JupyterLab tutorial.

## Expected behavior

Advanced JSON setting editor always pops up.

## Context

- Operating System and version: MacOS 12
- Browser and version: Firefox 99
- JupyterLab version: 3.3.4

I don't see any javascript errors in the browser console.

CC @pya

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fixes #12485

## Code changes

Ensure settings editor is attached before activating it.

## User-facing changes

The settings editor should always launch.

## Backwards-incompatible changes

N/A

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -21,7 +14,7 @@
   resolved "https://registry.npmjs.org/@babel/compat-data/-/compat-data-7.17.7.tgz#078d8b833fbbcc95286613be8c716cef2b519fa2"
   integrity sha512-p8pdE6j0a29TNGebNm7NzYZWB3xVZJBZ7XGs42uAKzQo8VQ3F0By/cQCtUEABwIqw5zo6WA4NbmxsfzADzMKnQ==
 
-"@babel/core@7.16.12":
+"@babel/core@7.16.12", "@babel/core@^7.1.0", "@babel/core@^7.10.2", "@babel/core@^7.7.5":
```
Comment (jtpio) [inline yarn.lock:17]: Is the `yarn.lock` update related to the change in this PR?

Comment (fcollonval) [reply yarn.lock:17]: No it is a clean up that was applied by yarn. I can definitely drop it from this PR.

Comment (jtpio) [reply yarn.lock]: Thanks. Maybe we can do the cleanup right after in a separate PR.

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
@meeseeksdev please backport to 3.4.x

**Pull request merged**